### PR TITLE
Add modelsToIgnore option

### PR DESCRIPTION
**Pull request by aquid**
… contains a list of models that are to be prevented from being published in loopback angular SDK.

cc @bajtos

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by bajtos** [comment]
@slnode ok to test

File: lib/services.js [hunk 1]
```diff
@@ -48,12 +48,14 @@ module.exports = function generateServices(app, options) {
     options = {
       ngModuleName: arguments[1],
       apiUrl: arguments[2],
+      modelsToIgnore: arguments[3] ? arguments[3] : []
```
Comment (bajtos) [inline lib/services.js]: This branch is for the legacy API that should not be extended by new features. Please remove this line.

File: lib/services.js [hunk 2]
```diff
@@ -72,10 +74,19 @@ module.exports = function generateServices(app, options) {
...
+  var modelsToIgnore = options.modelsToIgnore;
+
   app.handler('rest').adapter.getClasses().forEach(function(c) {
     var name = c.name;
     c.description = c.sharedClass.ctor.settings.description;
 
+    if(modelsToIgnore.indexOf(name) >= 0){
+      // Skip classes that are provided in options.modelsToIgnore array
+      // We don't want to publish these models in angular app
+      console.error('Skipping %j model as it is not to be published', name);
```
Comment (bajtos) [inline lib/services.js]: IMO, `console.error` should be used to log error messages, which is not the case here. Why is it needed to print a message when we are just following the instructions provided by the user?

**Comment by bajtos** [comment]
@aquid also please add a unit-test to verify your new feature and prevent regressions in the future. You can inspire by the tests added by https://github.com/strongloop/loopback-sdk-angular/commit/61b36412bb02f4153d30ca797b068c065e7eb45f.

File: lib/services.js (hunk 2 above)
Comment (superkhau) [inline lib/services.js]: console.warn

Comment (aquid) [inline lib/services.js]: @bajtos and @superkhau Thanks for this. I will update it with `console.warn` or maybe `console.log`. I did this to highlight the log across all other logs. But definitely that's not the right way and I should change it.

**Comment by pulkitsinghal** [comment]
@bajtos - please review this https://github.com/strongloop/loopback-sdk-angular/pull/226 before Aquid gets going otherwise he'll also be fixing stuff and not making checkins that are purely related to his theme

**Comment by aquid** [comment]
Made the changes to remove `console.error` to `console.warn` and added unit test for testing the feature.
https://github.com/strongloop/loopback-sdk-angular/pull/224/commits/dec5b345189509da68cd0594335e957b797298c0

cc @bajtos @superkhau @pulkitsinghal

**Comment by pulkitsinghal** [comment]
Hey @bajtos, can you onec agai nask the system to:
> @slnode ok to test

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: test.e2e/spec/services.spec.js [hunk 3]
```diff
@@ -1090,6 +1090,32 @@ define(['angular', 'given', 'util'], function(angular, given, util) {
         });
       });
     });
+
+    describe('$resource not generated for ignonred models', function() {
```
Comment (bajtos) [inline test.e2e/spec/services.spec.js]: typo: `ignonred`, should be `ignored`.

File: test.e2e/test-server.js [hunk 4]
```diff
@@ -114,6 +115,15 @@ masterApp.post('/setup', function(req, res, next) {
           apiUrl: apiUrl,
           includeSchema: includeSchema,
         });
+      } else if (modelsToIgnore) {
```
Comment (bajtos) [inline test.e2e/test-server.js]: This way, `modelsToIgnore` and `includeSchema` cannot be use together.

I am proposing a different solution:

```js
if (includeSchema || modelsToIgnore) {
  // the new options-based API
  servicesScript = generator.services(lbApp, {
    ngModuleName: name,
    apiUrl: apiUrl,
    includeSchema: includeSchema,
    modelsToIgnore: modelsToIgnore,
  });
} else {
  // the old API...
}
```

**Comment by bajtos** [comment]
Left two comments. @aquid After you address them, could you please rebase your patch on top of the latest `master` and squash all commits into a single one, with a descriptive commit message following [50/27 rule](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by aquid** [comment]
@bajtos Updated the PR with new commits. Since I had some rebasing issues, a new PR has been generated( #235 ).

Please review it and let me know.
cc @pulkitsinghal

**Pull request closed without merging**